app/app/(dashboard)/organization/[id]/audience/page.tsx

```diff
@@ -20,14 +20,23 @@ export default async function AudienceList({
 
   const orgId = decodeURIComponent(params.id);
 
+  // Check if user is a member of this organization
+  const member = await (prisma as any).organizationMember.findUnique({
```

Comment:
Seems like we do a lot of organization checks, might want to move this to a reusable decorator or component of some sort.